third_party/nvidia/backend/compiler.py

```diff
@@ -128,6 +130,8 @@ def make_ttgir(mod, metadata, opt, capability):
         passes.ttir.add_convert_to_ttgpuir(pm, opt.num_warps, 32, opt.num_ctas, capability)
         # optimize TTGIR
         passes.ttgpuir.add_coalesce(pm)
+        if capability // 10 >= 8:
+            passes.ttgpuir.add_fp32_dot_tc(pm)
```

Comment:
What happens if I pass "3xtf32" but I'm not on `capability // 10 >= 8`?

Reply:
We fallback transparently to `ieee`, same as we do now with `allow_tf32=True` now. Added to the docs in TritonOps.td and the public docs.

Reply:
Also Phil (who's the boss here) said

> And of course each target should also have a supported_fp32_backend attribute so that users get an error if they try to use tf32 on AMD GPUs.

so it seems to me like one could extend this logic to different nvidia GPUs just as easily as to nvidia vs AMD.

Reply:
Note that we already "transparently downgrade" `allow_tf32=True` to `allow_tf32=False` if there are no TC.

About `"foobar"`, that'll throw when we use enums. I'll push that implementation soon.